core/src/hlist.rs

```diff
@@ -93,9 +102,10 @@ pub trait HList: Sized {
     /// assert_eq!(h.len(), 2);
     /// # }
     /// ```
+    #[deprecated(since = "0.5.0", note = "Please use <Self as HList>::Len::[USIZE | U8 | U32 | ... ] instead")]
```

Comment:
I think it would not make much sense to deprecate this method and push users towards using the Typenum-coupled type.